File: security/core/src/main/java/me/kzv/core/service/UserService.java
package me.kzv.core.service;

import lombok.RequiredArgsConstructor;
import me.kzv.core.domain.Account;
import me.kzv.core.repository.UserRepository;
import org.springframework.stereotype.Service;

@RequiredArgsConstructor
@Service
public class  UserService {

    private final UserRepository userRepository;

    public void createUser(Account account) {
        userRepository.save(account);
    }
}

File: hello-kotlin/kotlin-board/src/main/kotlin/me/kzv/board/service/UserService.kt
package me.kzv.board.service

import me.kzv.board.entity.User
import me.kzv.board.repository.UserRepository
import me.kzv.board.controller.dto.SignInRequestDto
import me.kzv.board.controller.dto.SignUpRequestDto
import org.springframework.stereotype.Service
import java.lang.RuntimeException

@Service
class UserService(val userRepository: UserRepository) {

    fun signUp(dto: SignUpRequestDto): Long {
        val user = User(userId = dto.userId, password = dto.password, name = dto.name, age = dto.age)
        return userRepository.save(user).id!!
    }

    fun signIn(dto: SignInRequestDto): Long {
        return userRepository.findByUserIdAndPassword(dto.userId, dto.password)
                .orElseThrow { RuntimeException() }.id!!
    }
}
